src/annotation/askui-annotation-step-reporter.ts

```diff
@@ -0,0 +1,80 @@
+import fs from "fs";
+import { Step, Reporter, Annotation, ReporterConfig } from "askui";
+import path from "path";
+import { JSDOM } from 'jsdom';
+
+export enum AnnotationLevel {
+  DISABLED = 'disabled',
+  ON_FAILURE = 'onFailure',
+  ALL = 'all',
+}
+
+export interface AnnotationReporterConfig extends ReporterConfig {
+  annotationLevel: AnnotationLevel;
+  folderPath: string;
+  fileNameTemplate: string;
+}
+
+export class AskUIAnnotationStepReporter implements Reporter {
+
+  reporterConfig: AnnotationReporterConfig = {
+    annotationLevel: AnnotationLevel.DISABLED,
+    folderPath: 'report',
+    fileNameTemplate: '_testStep_annotation',
+  }
+
+  config: ReporterConfig = {
+    withScreenshots: 'always',
+    withDetectedElements: 'always',
+  }
+
+  constructor(reporterConfig: AnnotationReporterConfig) {
+    if (reporterConfig.annotationLevel !== undefined) {
+      this.reporterConfig.annotationLevel = reporterConfig.annotationLevel;
+    }
+    if (reporterConfig.fileNameTemplate !== undefined) {
+      this.reporterConfig.fileNameTemplate = reporterConfig.fileNameTemplate;
+    }
+    if (reporterConfig.folderPath !== undefined) {
+      this.reporterConfig.folderPath = reporterConfig.folderPath;
+    }
+    if (reporterConfig.withScreenshots !== undefined) {
+      this.config.withScreenshots = reporterConfig.withScreenshots;
+    }
+    if (reporterConfig.withDetectedElements !== undefined) {
+      this.config.withDetectedElements = reporterConfig.withDetectedElements;
+    }
```

Comment:
Would remove that because you simply need to set `withDetectedElements` based on how `annotationLevel` is set from my perspective and because the corresponding values are always going to be `undefined` in the param object passed...